panacea/grpc_client.go

```diff
@@ -84,3 +133,109 @@ func (c *GrpcClient) GetDeal(id string) (markettypes.Deal, error) {
 
 	return *response.GetDeal(), nil
 }
+
+// RegisterDataValidator registers data validator on blockchain
+func (c *GrpcClient) RegisterDataValidator(address, endpoint string) error {
+	_, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
```

Comment:
You created a `ctx` here but it's not used anywhere. If you wanted to set the network timeout for broadcasting a tx, you can pass this `ctx` to the `BroadcastTx()` at line 223. If you wanted to set the overall timeout for entire jobs in this functions, you should pass this `ctx` to `GetAccountNumber()` and `GetSequence()` as well as `BroadcastTx()`.

But, IMHO, we don't need to make all complicated. It would be enough to set timeout only for `BroadcastTx()`. Then, you should set the timeout bigger than 6s (which is an average block time).

Reply:
Ah, okay. I have a question. Does `time.Second` is proper time to `BroadcastTx()`?

Reply:
more than `6 * time.Second` because the average block time is 6s.

Reply:
Okay, I just apply `6*time.Second` now.